ocean_provider/validation/algo.py

```diff
@@ -270,6 +270,10 @@ def validate(self):
 
         self.service = self.asset.get_service_by_id(self.data["serviceId"])
 
+        if not self.service:
```

Comment:
Moved this check before `check_asset_consumable` to prevent key errors when accessing `self.service.service_endpoint`